backend/app/usecase/repository/shortlink_fake.go

```diff
@@ -12,6 +12,14 @@ var _ ShortLink = (*ShortLinkFake)(nil)
 // ShortLinkFake accesses ShortLink information in short_link table through SQL.
 type ShortLinkFake struct {
 	shortLinks map[string]entity.ShortLink
+	// TODO(issue#958) use eventbus for propagating short link change to user short link repo
+	userShortLinkRepoFake *UserShortLinkFake
+}
+
+// ProvideUserShortLinkRepoFake allows ShortLinkFake to propagate new alias to user short link repo.
```

Comment:
It's a dependency for `ShortLinkFake`. Can we inject it through the constructor? If you just trying to fix this test. I am okey with keeping what you have for now.

Reply:
Right, I can just set it inside `NewShortLinkFake`. 👍